### PR TITLE
Make section numbering consistent

### PR DESCRIPTION
**Pull request by randallreedjr**
* Renumber items where numbers were skipped

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,8 +154,8 @@ Other Style Guides
     const item = {};
     ```
 
-  <a name="es6-computed-properties"></a><a name="3.4"></a>
-  - [3.4](#es6-computed-properties) Use computed property names when creating objects with dynamic property names.
+  <a name="es6-computed-properties"></a><a name="3.2"></a>
```
Comment (ljharb) [inline README.md]: the anchor links are intentionally left as they are, to ensure old links continue to work forever. same throughout.

File: README.md [hunk 2]
```diff
@@ -1260,7 +1260,7 @@ Other Style Guides
 
     > Why? They don't transpile well to ES5.
 
-  <a name="generators--spacing"></a>
+  <a name="generators--spacing"></a><a name="11.3"></a>
```
Comment (ljharb) [inline README.md]: please do not add any new links that have numbers. that's the old-style legacy format, and we won't be adding or changing numbered links forevermore.

File: README.md (hunk 1 above)
Comment (randallreedjr) [inline README.md]: Thanks for pointing this out @ljharb. Does this mean that you would prefer to keep the numbering as is (out of sequence), or update the numbering but leave the links alone?

File: README.md (hunk 2 above)
Comment (randallreedjr) [inline README.md]: Ah, I suspected that might be the case, but couldn't confirm.

File: README.md (hunk 1 above)
Comment (randallreedjr) [inline README.md]: @ljharb In that case, I think this PR is ready to be reviewed again.

**Pull request merged**